lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -74,6 +75,66 @@ public void auth_client_match_tokendetails_null_clientId() {
         }
     }
 
+    /**
+     * RSA4d:  If a request by a realtime client to an authUrl results in an HTTP 403 response,
+     * or any of an authUrl request, an authCallback, or a request to Ably to exchange
+     * a TokenRequest for a TokenDetails result in an ErrorInfo with statusCode 403,
+     * as part of an attempt by the realtime client to authenticate, then the client library
+     * should transition to the FAILED state, with an ErrorInfo (with code 80019, statusCode 403,
+     * and cause set to the underlying cause) emitted with the state change and set as the connection
+     * errorReason
+     *
+     * Verify end connection state is failed
+     * Spec: RSA4d, RSA4d1
+     */
+    @Test
+    public void auth_client_fails_authorize_server_forbidden() {
+        try {
+            ClientOptions optsForToken = createOptions(testVars.keys[0].keyStr);
+            AblyRest ablyForToken = new AblyRest(optsForToken);
+            TokenDetails tokenDetails = ablyForToken.auth.requestToken(null, null);
+
+            ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+            opts.autoConnect = false;
+            opts.tokenDetails = tokenDetails;
+            opts.useTokenAuth = true;
+            opts.authUrl = "https://echo.ably.io/respondwith";
+            opts.authParams = new Param[]{ new Param("status", 403)};
+
+            final AblyRealtime ablyRealtime = new AblyRealtime(opts);
+            ablyRealtime.connection.connect();
+
+            Helpers.ConnectionWaiter connectionWaiter = new Helpers.ConnectionWaiter(ablyRealtime.connection);
+            connectionWaiter.waitFor(ConnectionState.connected);
+
+            ablyRealtime.connection.once(ConnectionEvent.failed, new ConnectionStateListener() {
+                @Override
+                public void onConnectionStateChanged(ConnectionStateChange stateChange) {
+                    assertEquals(stateChange.previous, ConnectionState.connected);
+                    assertEquals(stateChange.reason.code, 80019);
```

Comment:
This should follow the right format, `expected` and `actual` values should be at the right places